src/Symfony/Component/HttpFoundation/HeaderBag.php

```diff
@@ -105,23 +105,41 @@ public function add(array $headers)
      * @param string|null $default The default value
      * @param bool        $first   Whether to return the first value or all header values
      *
-     * @return string|string[]|null The first header value or default value if $first is true, an array of values otherwise
+     * @return string|string[]|int|null The first header value or default value if $first is true, an array of values otherwise
      */
     public function get($key, $default = null, $first = true)
+    {
+        // do we trigger an deprecation ?
+        // @trigger_error(sprintf('%s::%s is deprecated since Symfony 4.4, use %s::getFirst() or %s::getAll() instead.', __CLASS__, __METHOD__, __CLASS__, __CLASS__), E_USER_DEPRECATED);
+        if (true === $first) {
+            return $this->getFirst($key, $default);
+        }
+
+        return $this->getAll($key, $default);
+    }
+
+    public function getFirst($key, $default = null)
```

Comment:
we should aim for `?string $default = null` and `: ?string`

Reply:
we can do : ? string because it can return int :/